src/si/fmt.rs

```diff
@@ -0,0 +1,133 @@
+///
+/// # Usage
+/// ```
+/// # use uom::si::Unit;
+/// # use uom::si::{f32::*, length::{centimeter, meter}, fmt::{Arguments, DisplayStyle::*}};
+/// let length = Length::new::<meter>(1.0);
+/// let a = Arguments::new(centimeter, Description);
+/// assert_eq!("100 centimeters", format!("{}", a.with(length)));
+/// ```
+#[allow(missing_debug_implementations)] // Prevent accidental direct use.
+#[derive(Clone)]
+pub struct Arguments<T: Unit> {
+    subunit: T,
+    style: DisplayStyle,
+}
+
+impl<T: Unit> Arguments<T> {
+
+    /// Creates a new `Argument` for displaying a quantity in the specified units and style.
+    pub fn new(subunit: T, style: DisplayStyle) -> Self {
+        Self { subunit, style }
+    }
+
+    /// Specifies a quantity to display.
+    pub fn with<D, U, V>(&self, quantity: Quantity<D, U, V>) -> QuantityArguments<T, D, U, V>
+    where
+        D: Dimension + ?Sized,
+        U: Units<V> + ?Sized,
+        V: Num + Conversion<V>,
+    {
+        QuantityArguments {
+            arguments: self.clone(), // Allow use of this format style with multiple quantities
+            quantity,
+        }
+    }
+
+}
+
+#[doc(hidden)]
+pub struct QuantityArguments<T: Unit, D: Dimension + ?Sized, U: Units<V> + ?Sized, V: Num + Conversion<V>> {
+    arguments: Arguments<T>,
+    quantity: Quantity<D, U, V>,
+}
+
+trait Convert<V> { // Use this get<N> instead of the (private) get<N> implemented per-unit.
+    fn get<N: Unit + Conversion<V>>(&self, _unit: N) -> V;
+}
+
+impl<D, U, V> Convert<V> for Quantity<D, U, V>
+where
+    D: Dimension + ?Sized,
+    U: Units<V> + ?Sized,
+    V: Num + Conversion<V> + Clone,
+{
+    fn get<T: Unit + Conversion<V>>(&self, _unit: T) -> V {
+        self.value.clone() / T::conversion().value()
+    }
+}
+
+macro_rules! display_arguments {
+    ($name:ident, $pat:expr) => {
+        impl<T, D, U, V> fmt::$name for QuantityArguments<T, D, U, V>
+        where
+            T: Unit + Conversion<V>,
+            D: Dimension + ?Sized,
+            U: Units<V> + ?Sized,
+            V: Num + Conversion<V> + fmt::$name + Clone,
+        {
+            fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                let value = self.quantity.get::<T>(self.arguments.subunit);
+                write!(f, $pat, value)?;
+                write!(f, " {}", match self.arguments.style {
+                    DisplayStyle::Abbreviation => T::abbreviation(),
+                    DisplayStyle::Description => if value.is_one() {T::singular()} else {T::plural()},
+                })
+            }
+        }
+    };
+}
+
+display_arguments!(Binary, "{:b}");
+display_arguments!(Debug, "{:?}");
+display_arguments!(Display, "{}");
+display_arguments!(LowerExp, "{:e}");
+display_arguments!(LowerHex, "{:x}");
+display_arguments!(Octal, "{:o}");
+display_arguments!(UpperExp, "{:E}");
+display_arguments!(UpperHex, "{:X}");
+
+#[cfg(test)]
+mod tests {
+
+    use si::fmt::{Arguments, DisplayStyle::*};
```

Comment:
`uom` still supports `rustc` back to 1.20.0 which doesn't support nested imports.